### DISCUSSION TITLE
Can't build Dockerfile with gauge for arm64 architecture

### DISCUSSION THREAD

**Discussion by gadgadskiy** [getgauge/gauge#2227] [General]
Hi. I have a working Dockerfile to build an image for running gauge tests in the docker container for intel architecture. Now, I'm trying to make a similar file with the version for Mac with an m1 processor but get a mistake:
`The command '/bin/sh -c gauge config check_updates false' returned a non-zero code: 2`
This mistake appears on each mention of gauge commands, like `gauge config check_updates false`, `gauge -v`, etc.
The simplest Dockerfile to reproduce the error:
```
FROM maven:3.6-openjdk-11

ARG GAUGE_VERSION=1.4.3

ENV GAUGE_ARCHIVE gauge-$GAUGE_VERSION-darwin.arm64.zip
ENV GAUGE_TELEMETRY_ENABLED false

RUN wget --no-check-certificate --quiet "https://github.com/getgauge/gauge/releases/download/v$GAUGE_VERSION/$GAUGE_ARCHIVE"
RUN unzip $GAUGE_ARCHIVE -d /usr/local/bin
RUN rm -rf $GAUGE_ARCHIVE
RUN gauge config check_updates false
RUN ls /usr/local/bin/gauge

ENTRYPOINT ["mvn"]
```

The same Dockerfile builds successfully when I've changed version bach to 1.4.3-linux.x86_64.zip.
First, I thought it happened because I had run the script on a machine with an Intel processor. But on a device with an m1 processor, I've got a similar error too.

What I'm doing wrong?

**Comment by zabil** [comment]
I don't think Darwin binaries can be used inside Docker because `FROM maven:3.6-openjdk-11` is an image that is linux x86_64. So please use 1.4.3-linux.x86_64.zip instead.

**Reply by gadgadskiy** [reply]
@zabil, so is there no possibility of creating a docker container with such an arm64 image?

**Comment by sriv** [comment]
There are a couple of things:

1. you need to use [`armv8/maven`](https://hub.docker.com/r/arm64v8/maven/) images for docker on arm64.
2. gauge does not (yet) have arm64 binaries for linux.

I used the multistage build to create a gauge-docker image on m1 mac. Please see if this works for you:

```
FROM arm64v8/golang as build

ARG GAUGE_VERSION=1.4.3

RUN go install github.com/getgauge/gauge@v$GAUGE_VERSION

FROM arm64v8/maven:3.6-openjdk-11

ENV GAUGE_TELEMETRY_ENABLED false

COPY --from=build /go/bin/gauge /usr/local/bin/
RUN gauge config check_updates false
ENTRYPOINT ["mvn"]
```

**Reply by gadgadskiy** [reply]
It is working for me. Thank you!
